src/Ztm.WebApi/Controllers/TokenController.cs

```diff
@@ -0,0 +1,132 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using NBitcoin;
+using Ztm.Configuration;
+using Ztm.WebApi.Callbacks;
+using Ztm.WebApi.Models;
+using Ztm.WebApi.Watchers.TransactionConfirmation;
+using Ztm.Zcoin.NBitcoin.Exodus;
+using Ztm.Zcoin.Rpc;
+using Transaction = Ztm.WebApi.Models.Transaction;
+
+namespace Ztm.WebApi.Controllers
+{
+    [Route("api")]
+    [ApiController]
+    public class TokenController : ControllerBase
+    {
+        readonly IZcoinRpcClientFactory factory;
+        readonly IConfiguration configuration;
+        readonly ITransactionConfirmationWatcher watcher;
+        readonly ICallbackRepository callbackRepository;
+        readonly IRuleRepository ruleRepository;
+
+        public TokenController(
+            IZcoinRpcClientFactory factory,
+            IConfiguration configuration,
+            ITransactionConfirmationWatcher watcher,
+            ICallbackRepository callbackRepository,
+            IRuleRepository ruleRepository)
+        {
+            if (factory == null)
+            {
+                throw new ArgumentNullException(nameof(factory));
+            }
+
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            if (watcher == null)
+            {
+                throw new ArgumentNullException(nameof(watcher));
+            }
+
+            if (callbackRepository == null)
+            {
+                throw new ArgumentNullException(nameof(callbackRepository));
+            }
+
+            if (ruleRepository == null)
+            {
+                throw new ArgumentNullException(nameof(ruleRepository));
+            }
+
+            this.factory = factory;
+            this.configuration = configuration;
+            this.watcher = watcher;
+            this.callbackRepository = callbackRepository;
+            this.ruleRepository = ruleRepository;
+        }
+
+        [HttpPost("issue-tokens")]
+        public async Task<IActionResult> Issue([FromBody] Issuing issueing)
+        {
+            var zcoinConfig = this.configuration.GetZcoinSection();
+
+            using (var client = await this.factory.CreateRpcClientAsync(CancellationToken.None))
+            {
+                var property = new Property(zcoinConfig.Property.Id, zcoinConfig.Property.Type);
+
+                var tx = await client.GrantPropertyAsync
+                (
+                    property,
+                    zcoinConfig.Property.Distributor.Address,
```

Comment:
This is not correct. For token issuing we will always grant from `issuer` to `distributor`.